src/dmd/backend/cgcod.c

```diff
@@ -1128,25 +1128,32 @@ void stackoffsets(int flags)
     for (int si = 0; si < globsym.top; si++)
     {   symbol *s = globsym.tab[si];
 
-        if (s->Sisdead(anyiasm))
+        /* Don't allocate space for dead or zero size parameters
+         */
+        switch (s->Sclass)
         {
-            /* The variable is dead. Don't allocate space for it if we don't
-             * need to.
-             */
-            switch (s->Sclass)
-            {
-                case SCfastpar:
-                    if (!(funcsym_p->Sfunc->Fflags3 & Ffakeeh))
-                        continue;   // don't need consistent stack frame
-                    break;
+            case SCfastpar:
+                if (!(funcsym_p->Sfunc->Fflags3 & Ffakeeh))
+                    goto Ldefault;   // don't need consistent stack frame
+                break;
 
-                case SCshadowreg:
-                case SCparameter:
-                    break;          // have to allocate space for parameters
+            case SCparameter:
+                if (type_parameterSize(s->Stype) == 0)
+                {
+                    Para.offset = _align(REGSIZE,Para.offset); // align on word stack boundary
+                    s->Soffset = Para.offset;
+                    continue;
```

Comment:
So how does passing around an empty aggregate change here?  I'd expect the equivalent of replacing `foo(s)` with `foo({})`.
That is to say that empty aggregate vars and params aren't passed around directly - having nothing themselves to pass around - they are just an alias for a field-less struct literal.

Reply:
I did some experiments. You can take the address of a parameter that is an empty struct. It is the start of the next parameter. So it definitely has an offset.